index.src.html

```diff
@@ -1146,6 +1146,48 @@ <h3 id="gc">Garbage Collection</h3>
   removed from the <a>report buffer</a> of any <a>reporting observer</a>.
 </section>
 
+<section>
+  <h2 id="deployment">Deployment Considerations</h2>
+
+  <h3 id="custom-metadata">Custom Metadata</h3>
+
+  A server might want to include additional metadata in reports that are
+  generated for their origin.  This can be accomplished by encoding the extra
+  metadata in the <a for="ReportTo">`url`</a> of any <a
+    for="ReportTo">`endpoints`</a> in the <a>`Report-To`</a> response headers
+  for the origin — for example, in the URL path or query parameters.
+
+  <pre>
+    <a>Report-To</a>: { "<a for="ReportTo">group</a>": "csp",
+                 "<a for="ReportTo">max-age</a>": 10886400,
+                 "<a>endpoints</a>": [
+                   { "<a for="ReportTo">url</a>": "https://example.com/reports?nonce=e897932f" }
+                 ] }
+  </pre>
+
+  Since the instructions in a <a>`Report-To`</a> header will be used for future
+  requests to the same origin, the server SHOULD NOT use this mechanism to
+  encode metadata that is only valid for the current request.  The metadata MUST
```

Comment:
We can't enforce the `MUST` here, we can only provide warnings and recommendations. On that note, it may be worth considering moving this entire section into a non-normative appendix? 

> valid for all requests to the same origin from the same user.

Same user may or may not be relevant here.. For example, if you want to communicate deployed version, then that has nothing to do with the user. Also, it's possible that the reported metadata will be out of date, since the report could be triggered due to a failed navigation without seeing the most up to date report-to information.

Reply:
> Also, it's possible that the reported metadata will be out of date

Right, that's what I was getting at with the user part — if the metadata is valid for all requests made by that user, then even if they use an old header to report on a failed request, the metadata is still valid.

I like your out-of-date wording, better, though — describe the facts, let the reader / server owner decide how that affects their requirements.

If you want to encode something in the upload URL, though, you have to make sure that you deploy the same version to the same user for all

Reply:
Hmm, as a thought experiment.. Instead of talking about "same version", perhaps focus on "last seen reporting endpoint wins" and you should take that into account — e.g. this is not the right mechanism to execute a/b tests, or similar.

Reply:
The fact that out-of-date URLs can be used to report network failures (as the more recent `Report-To` never made it to the browser) means that it'd be very hard to avoid replay attacks when aggregating reports on the server.
If we are limiting reports to the original IP on which the `Report-To` response was received, it raises the bar here (e.g. attacker would need to be able to reliably spoof that IP), but only if the server e.g. deletes reports which responses got RST.

Reply:
Thinking about it some more, combining per-IP nonces with per-IP rate-limiting (so that a certain IP can't report more than what a reasonable user would) can probably overcome replays, or at least make them less impactful.